### PR TITLE
Harvey style assertions

### PR DESCRIPTION
**Pull request by nbanmp**

### CODE REVIEW

File: mythril/analysis/modules/user_assertions.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            message = eth_abi.decode_single(
+                "string",
+                bytes(
+                    state.mstate.memory[
+                        mem_start.value + 32 : mem_start.value + size.value
+                    ]
+                ),
+            ).decode("utf8")
+
+        description_head = "A user-provided assertion failed. Make sure the user-provided assertion is correct."
```
Comment (b-mueller) [inline mythril/analysis/modules/user_assertions.py]: I'd reduce `head` to "A user-provided assertion failed.".

File: mythril/analysis/modules/user_assertions.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+"""
+
+assertion_failed_hash = (
+    0xB42604CB105A16C8F6DB8A41E6B00C0C1B4826465E8BC504B3EB3E88B3E6A4A0
+)
+
+
+class UserAssertions(DetectionModule):
+    """This module searches for low level calls (e.g. call.value()) that
```
Comment (b-mueller) [inline mythril/analysis/modules/user_assertions.py]: Update class description

File: mythril/laser/ethereum/instructions.py [hunk 3]
```diff
@@ -1624,7 +1624,7 @@ def log_(self, global_state: GlobalState) -> List[GlobalState]:
         # TODO: implement me
         state = global_state.mstate
         dpth = int(self.op_code[3:])
-        state.stack.pop(), state.stack.pop()
+        mem_start, size = state.stack.pop(), state.stack.pop()
```
Comment (norhh) [inline mythril/laser/ethereum/instructions.py]: We don’t need this change atm right?(that sonar will raise some code smells for unused code)

File: mythril/analysis/modules/user_assertions.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        annotation = get_potential_issues_annotation(state)
+        annotation.potential_issues.extend(potential_issues)
+
+    def _analyze_state(self, state: GlobalState):
+        """
+
+        :param state:
+        :return:
+        """
+        mem_start = state.mstate.stack[-1]
```
Comment (norhh) [inline mythril/analysis/modules/user_assertions.py]: we can write these 3 in a single like I think.
`topic, size, mem_start = state.mstate.stack[-3:]`

**Pull request merged**